sdv/metadata/multi_table.py

```diff
@@ -579,6 +599,14 @@ def _validate_single_table(self, errors):
                 error = str(error).replace(
                     'The following errors were found in the metadata:\n', title)
                 errors.append(error)
+                try:
+                    foreign_key_cols = [
+                        relationship['child_foreign_key'] for relationship in self.relationships
+                        if table_name == relationship.get('child_table_name')
+                    ]
+                    self._validate_column_relationships_foreign_keys(table, foreign_key_cols)
+                except Exception as col_relationship_error:
```

Comment:
why does this need to be nested in the previous except? Also, can we calculate the foreign_key_cols once outside the for loop?